anxcloud/utils.go

```diff
@@ -77,3 +77,25 @@ func listAllPages[T any](pageRetriever func(int) ([]T, error)) ([]T, error) {
 
 	return allPages, nil
 }
+
+func sliceSubstract[T comparable](a, b []T) []T {
+	out := make([]T, 0, len(a))
+outer:
+	for i := range a {
+		for j := range b {
+			if a[i] == b[j] {
+				continue outer
+			}
+		}
+		out = append(out, a[i])
+	}
+	return out
+}
+
+func castInterfaceArray[T any](in []interface{}) []T {
```

Comment:
please either return an error if any cast is impossible or rename to `mustCastInterfaceArray`

Reply:
Decided for `mustCastInterfaceArray` as underlying type is validated by schema.